src/main/java/reactor/netty/http/client/HttpClientOperations.java

```diff
@@ -558,13 +560,22 @@ protected void onInboundNext(ChannelHandlerContext ctx, Object msg) {
 					ReferenceCountUtil.release(msg);
 					return;
 				}
+			} else {
+				redirected = true;
 			}
+
 			if (msg instanceof FullHttpResponse) {
 				super.onInboundNext(ctx, msg);
 				terminate();
 			}
 			return;
 		}
+
+		if(redirected) {
+			ReferenceCountUtil.release(msg);
+			return;
```

Comment:
Why don't you release the response when `notRedirected` is invoked, just before emitting the RedirectClientException?

Reply:
Because `HttpResponse` is not implementing `ByteBufHolder`, so we don't need to release it. What caused the leak is the `HttpContent` coming in a separate call (but belonging to the `HttpResponse` with the redirect), as it implements `ByteBufHolder`, so we need to release it. But it currently is not released when there is a redirect, and the content is not empty.

Reply:
just wondering whether it can be FullHttpResponse

Reply:
I have no idea in what case we receive a FullHttpResponse. According to the code, I guess it's possible that we receive it for a redirect with content, so probably I should update the code to take care of this case.

Reply:
Well, I've looked through the code and I can see no way that this piece of code is reached. In the channel pipeline, `HttpResponseDecoder`, which is the previous handler produces a `FullHttpResponse` in only one case, when there is an invalid response, but in that case there is no content free up. I've also ran all the tests and not a single one covers this branch. 

So, to conclude, on second thought I would leave the PR as it is, because I see no point in handling this case.

Reply:
ok